### PR TITLE
lint code-templates

### PR DESCRIPTION
**Pull request by DanielGluskin**
- fix lint errors on /code-templates
- some rules were set to "warn" or "off" in order to remove linting failure so we can run it inside the CI.

Next steps:
- determine which rules (and level) should we use.
- add linting as a part of the CI.

### CODE REVIEW

**Review by rluvaton — APPROVED** [review]
Great job! thanks!

File: src/code-templates/package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "build": "tsc",
     "build:watch": "tsc --watch",
     "lerna": "lerna",
+    "lint": "eslint '**/*.{js,jsx,ts}'",
```
Comment (rluvaton) [inline src/code-templates/package.json:11]: I think we don't need jsx here

**Review by rluvaton — DISMISSED** [review]

File: src/code-templates/.prettierrc.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module.exports = {
   trailingComma: "es5",
   tabWidth: 2,
-  semi: false,
+  semi: true,
```
Comment (goldbergyoni) [inline src/code-templates/.prettierrc.js:4]: @DanielGluskin @rluvaton The main idea of Prettier is "we set good default so you don't deal with it"

Should we really express an opinion here?

File: src/code-templates/package.json (hunk 1 above)
Comment (goldbergyoni) [reply src/code-templates/package.json:11]: @rluvaton @DanielGluskin Should not we specify the includes/excludes in eslintrc (where all the definition is located) and here just specify "."?

File: src/code-templates/.prettierrc.js (hunk 2 above)
Comment (goldbergyoni) [inline src/code-templates/.prettierrc.js:4]: @DanielGluskin @rluvaton The main idea of Prettier is "we set good default so you don't deal with it"

Should we really express an opinion here?

**Review by goldbergyoni — CHANGES_REQUESTED** [review]
Looks awesome and important, left two questions/comments

File: src/code-templates/.prettierrc.js (hunk 2 above)
Comment (DanielGluskin) [reply src/code-templates/.prettierrc.js:4]: 1. yes in my opinion. Will correct.
2. It's a philosophic question. On one hand, we should try stick to the defaults. On the other, if we have the option to use a settings file, we should use it time-to-time. So the I think it's fine to edit some rules, if we mostly agree on them, like in this case.

**Review by goldbergyoni — APPROVED** [review]

**Comment by goldbergyoni** [comment]
@DanielGluskin We have linters finally! 💚

Few topics to discuss/enhance:

1. Many warnings - This affects the DX. IMO we should disable a rule or fix the code, zero warnings
2. Should we really support es2021 or es2020?
3. We should add linters to the root project
4. We should 'npm run lint' to our CI
5. There are popular and useful custom linters that we may consider like:
https://www.npmjs.com/package/eslint-plugin-promise
https://www.npmjs.com/package/eslint-plugin-jest
https://www.npmjs.com/package/eslint-plugin-import

**Pull request merged**